examples/graupnerbrunelstdp/stdputil.py

```diff
@@ -15,6 +15,22 @@
 
 logging.basicConfig(level=logging.WARN)
 
+#Note: having debug logging statements increases the run time by ~ 25%,
+# because they exist in tight loops, and expand their outputs, even when
+# debug is off, so we disable logging if possible.  Set this to true if
```

Comment:
Yeah, depending on what the logging level is, it does execute some code.  It bails out relatively early, but there is some overhead.

It looks like half the speedup comes from the logging_debug_vec not expanding the arrays, and the other half from forcing debug to be turned off in the tight loop.